lib/src/front_end/delimited_list_builder.dart

```diff
@@ -51,7 +51,16 @@ class DelimitedListBuilder {
 
   /// Creates the final [ListPiece] out of the added brackets, delimiters,
   /// elements, and style.
-  ListPiece build() {
+  Piece build() {
+    // To simplify the piece tree, if there are no elements, just return the
+    // brackets concatenated together.
```

Comment:
Double checking - is there any issue with empty lists containing comments?

Reply:
No, we don't. DelimitedListBuilder hoists comments up into separate elements, so `_elements` wouldn't be empty if there were any. Added a sentence here to that effect.